.github/workflows/kiali-ci.yml

```diff
@@ -32,7 +47,87 @@ jobs:
         run: make swagger-ci
 
       - name: Build
-        run: make clean build
+        run: make clean-all build
 
-      - name: Test
+      - name: Test backend
         run: make test-race
+
+      - name: Upload go binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: kiali
+          path: ~/go/bin/kiali
+
+  build_frontend:
+    name: Build and unit test frontend
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          cache: yarn
+          cache-dependency-path: frontend/yarn.lock
+
+      - name: Build
+        run: make clean-all build-ui
+
+      - name: Test frontend
+        run: |
+          cd frontend
+          yarn pretty-quick --check --verbose --branch origin/master
```

Comment:
does `origin/master` here assume this was triggered by a change to the master branch? Shouldn't this use the GITHUB_REF_NAME? Note that the GITHUB_REF might reference a PR branch name, so I'm not sure what the right thing to do is.

Reply:
Yes! Good eye